### ISSUE TITLE
CANINE model in huggingface transformers performs worse than mBERT?

### ISSUE THREAD

**Issue by cindyxinyiwang** [huggingface/transformers#13567]
I've been trying to use the newly added CANINE model from Google for multilingual NER task. Specifically, I finetune the CanineForTokenClassification model on WikiAnn English NER data. However, I got much worse performance on English NER compared to mBERT. Here are the rough performance numbers:
NER dev F1 on en:
canine: 79
mbert: 85
POS dev F1 on en:
canine: 90
mbert: 95

I also evaluated the finetuned model on the test data from a couple languages other than English, and they all seem worse than mBERT. I tried learning rate of 2e-5, 5e-5, and trained for 10, 20 epochs, but they don't seem to affect the results too much.
I'm not sure if anyone had the same result with me or if there are something wrong I did when using CANINE on huggingface transformers?

@stefan-it @NielsRogge 
## Environment info

- `transformers` version: 4.11.0.dev0
- Platform: Linux-4.15.0-133-generic-x86_64-with-glibc2.17
- Python version: 3.8.10
- PyTorch version (GPU?): 1.9.0 (True)

**Comment by stefan-it** [comment]
Hi @cindyxinyiwang ,

I ran a one-run experiment with Flair (I recently wrote an embedding implementation for CANINE to use it with Flair) and the results on XTREME/WikiANN Turkish split are:

| Model             | Dev F1 | Test F1
| ----------------- | ------ | -------
| mBERT (cased)     | 0.9358 | 0.9333
| `google/canine-s` | 0.9051 | 0.9044

So there's a current performance diff of -3.07%. I'll run another experiment with the `CANINE-C` model and report the results here!

**Comment by stefan-it** [comment]
I'm using:

```python
encoding = self.tokenizer([tokenized_string], padding="longest", return_tensors="pt")
hidden_states = self.model(**encoding).last_hidden_state
```

And the first - let's say character - embedding (analog. to first subword embedding).

**Comment by cindyxinyiwang** [comment]
I see thanks so much for sharing the results! My run of canine-c has similar results to canine-s. I think the huggingface implementation also uses the first character of each word. I thought it might be a sequence tagging issue. Then I additionally tried PAWS-X classification but it's also worse than mBERT on all languages...Not sure if there is anything specific we are missing

**Comment by stefan-it** [comment]
Hey @cindyxinyiwang what Pos Tagging dataset did you use btw. for English :thinking: E.g. in the datasets library there are a few English subsets, see [here](https://huggingface.co/datasets/viewer/?dataset=universal_dependencies).

**Comment by cindyxinyiwang** [comment]
Hey! I think i probably used all the concatenated English training data in UD 2.5. I used the preprocessing script from XTREME instead of directly from the datasets library

**Comment by cindyxinyiwang** [comment]
Anyways thanks for confirming with my observation. Maybe there is something wrong with the huggingface implementation, or perhaps there are some intricate fine-tuning hyperparameters...

**Comment by NielsRogge** [comment]
Thanks for trying out CANINE, I don't think i've made a mistake in my implementation, the model outputs the same tensors as the original one on the same data. Also, given that one can obtain an F1 score of > 0.90, I don't think it's a modeling issue. Note that in the original paper, the authors did not try out CANINE on NER problems, only QA.

Also cc'ing @dhgarrette, the original author.

**Comment by cindyxinyiwang** [comment]
I see that makes sense. Thanks so much for implementing the model so we can use it very easily!
